Orange/widgets/widget.py

```diff
@@ -46,21 +48,10 @@ class WidgetMetaClass(type(QDialog)):
     #noinspection PyMethodParameters
     # pylint: disable=bad-classmethod-argument
     def __new__(mcs, name, bases, kwargs):
-        from Orange.canvas.registry.description import (
-            input_channel_from_args, output_channel_from_args)
-
         cls = super().__new__(mcs, name, bases, kwargs)
         if not cls.name: # not a widget
             return cls
 
-        cls.inputs = [input_channel_from_args(inp) for inp in cls.inputs]
-        cls.outputs = [output_channel_from_args(outp) for outp in cls.outputs]
```

Comment:
It might be better to still leave these lines. From memory, I think I saw some widgets that self inspect their own .{inputs,outputs}. They might expect that they are 'normalized' as {Input,Output}Signal. `get_widget_description` might not be called if running widget standalone, or when the descriptions are loaded from cache.

In any case changing them in `get_widget_descriptions` will trip someone up sooner or later.